httpx/config.py

```diff
@@ -337,6 +340,79 @@ def __repr__(self) -> str:
         )
 
 
+class Retries:
+    """
+    Retries configuration.
+
+    Holds a retry limiting policy, and implements a configurable exponential
+    backoff algorithm.
+    """
+
+    def __init__(self, *retries: RetriesTypes, backoff_factor: float = None) -> None:
+        limits: RetriesTypes
+
+        if len(retries) == 0:
+            limits = RetryOnConnectionFailures(3)
+        elif len(retries) == 1:
+            limits = retries[0]
+            if isinstance(limits, int):
+                limits = (
+                    RetryOnConnectionFailures(limits) if limits > 0 else DontRetry()
+                )
+            elif isinstance(limits, Retries):
+                assert backoff_factor is None
+                backoff_factor = limits.backoff_factor
+                limits = limits.limits
+            else:
+                raise NotImplementedError(
+                    "Passing a `RetryLimits` subclass as a single argument "
+                    "is not supported. You must explicitly pass the number of times "
+                    "to retry on connection failures. "
+                    "For example: `Retries(3, MyRetryLimits(...))`."
+                )
+        elif len(retries) == 2:
+            default, custom = retries
+            assert isinstance(custom, RetryLimits)
```

Comment:
That might be a bit unclear at first, yes… Basically we want to allow `Retries()` OR `Retries(<int>)` OR `retries(<int>, <CustomRetryLimits>)`, but disallow `Retries(<CustomRetryLimits>)`, because that case is a bit ambiguous: do we mean "use the default number of retries", or "skip retrying on connection failures"? Hence the part in the docs about "explicitly passing the number of times to retry on connection failures".